app/controllers/floors_controller.rb

```diff
@@ -54,77 +49,37 @@ def floor_params
 
   private
 
-  def fetch_persistent_locations
-    Location.where(:persistent => true)
-  end
-
   def process_search(search_params)
-    @search_results = []
-    @search_results << Location.search_for(search_params)
-    @search_results << Tag.search_for(search_params)
-    @search_results << Trait.search_for(search_params)
-    @search_results
-  end
-
-  def normalize_search_result_floors(floors)
-    @normalized_floors = [nil, nil, nil, nil, nil, nil]
-    floors.each do |floor|
-      @normalized_floors[floor.level - 1] = floor
-    end
-    @normalized_floors
-  end
-
-  def extract_maps_from_floors(floors)
-    @maps = []
-    floors.each do |floor|
-      @maps << floor.map
-    end
-    @maps
+    return [] if search_params.nil?
+    results = []
+    results << Location.search_for(search_params)
+    results << Tag.search_for(search_params)
+    results << Trait.search_for(search_params)
+    results.flatten
   end
 
   def extract_floors(locations)
-    @search_result_floors = []
-      locations.each do |location|
-        if (location.is_a?(Location))
-          @search_result_floors << location.floor unless @search_result_floors.include?(location.floor)
-        elsif(locations.first.is_a?(Tag))
-          @search_result_floors << tag.location.floor unless @search_result_floors.include?(tag.location.floor)
-        else
-
-        end
-      end
-    normalize_search_result_floors(@search_result_floors)
-  end
-
-  def extract_icon_images
-    @icon_images = {}
-    Icon.all.each do |icon|
-      @icon_images[icon.id] = icon.icon_image
+    floors = Array.new(6)
+    locations.to_a.each do |location|
+      floors[location.floor.level - 1] = location.floor if floors[location.floor.level - 1].nil?
     end
-    @icon_images
+    floors
   end
 
   def extract_locations(search_results)
-    @locations = []
-    if search_results.first.is_a?(Trait)
-      search_results.each do |result|
-        @locations.concat(result.locations) if result.value == 'Yes'
-      end
-    elsif search_results.first.is_a?(Tag)
-      search_results.each do |result|
-        @locations << result.location
-      end
-    else
-      search_results.each do |result|
-        @locations << result
-      end
+    search_results.select do |r|
```

Comment:
Compacts this code, especially since there are 3 "search_results.each" pieces in the old code. Cuts down on code dupe.